test/controllers/test_password_expired_controller.rb

```diff
@@ -21,11 +21,35 @@ class Devise::PasswordExpiredControllerTest < ActionController::TestCase
     sign_in(@user)
   end
 
+  test 'redirects on show if user not logged in' do
+    sign_out(@user)
+    get :show
+    assert_redirected_to :root
+  end
+
+  test 'redirects on show if user does not need password change' do
+    @user.update(password_changed_at: Time.zone.now)
+    get :show
+    assert_redirected_to :root
+  end
+
   test 'should render show' do
```

Comment:
Does it make sense to add more context to match the other `test` strings?

```suggestion
  test 'should render show when user needs a password change' do
```